File: OOP based implmentation/OOP based implmentation/Project/objs/StudentAcademics.cpp
#include <iostream>
#include <fstream>
#include <string>
#include "StudentAcademics.h"
#include "../test/mcqs.h"

using namespace std;

void StudentAcademics::writeToStudentDetailsFile()
{
        ofstream writer("resources/StudentDetails.txt", ios_base::app);
        if (!writer.is_open())
        {
            cout << "An error occurred while writing to StudentDetails.txt." << endl;
            return;
        }
        writer << this->rollNum << "," << this->cgpa << "," << this->testScore << "," << this->skills << "," << this->expectedCTC << "\n";
        writer.close();
}

void StudentAcademics::writeToStudentTestFile()
{
        ofstream writer("resources/StudentTest.txt", ios_base::app);
        if (!writer.is_open())
        {
            cout << "An error occurred while writing to StudentTest.txt." << endl;
            return;
        }
        writer << this->rollNum << ":" << this->testScore << "\n";
        writer.close();
}

void StudentAcademics::writeToCSV()
{
        ofstream writer("resources/StudentAca.csv", ios_base::app);
        if (!writer.is_open())
        {
            cout << "An error occurred while writing to StudentAca.csv." << endl;
            return;
        }
        writer << this->rollNum << "," << this->cgpa << "," << this->testScore << "," << this->skills << "," << this->expectedCTC << "\n";
        writer.close();
}

StudentAcademics::StudentAcademics(const string &rollNum, float cgpa, const string &skills, float expectedCTC)
{
    this->rollNum = rollNum;
    this->cgpa = cgpa;
    this->skills = skills;
    this->expectedCTC = expectedCTC;
    this->testScore = MCQs::giveTest();

    writeToStudentDetailsFile();
    writeToStudentTestFile();
    writeToCSV();
}

string StudentAcademics::getRollNum() const
{
    return rollNum;
}

float StudentAcademics::getTestScore() const
{
    return testScore;
}

float StudentAcademics::getCgpa() const
{
    return cgpa;
}

string StudentAcademics::getSkills() const
{
    return skills;
}

float StudentAcademics::getExpectedCTC() const
{
    return expectedCTC;
}

File: OOP based implmentation/OOP based implmentation/Project/objs/StudentAcademics.java
package objs;

import java.io.BufferedWriter;
import java.io.FileWriter;
import java.io.IOException;
import test.mcqs;

public class StudentAcademics {
    private String rollNum;
    private float cgpa;
    private String skills;
    private float expectedCTC;
    private float testScore;

    // Constructor
    public StudentAcademics(String rollNum, float cgpa, String skills,float expectedCTC) {
        this.rollNum = rollNum;
        this.cgpa = cgpa;
        this.skills = skills;
        this.expectedCTC = expectedCTC;
        this.testScore = mcqs.giveTest(); 

        writeToStudentDetailsFile();
        writeToStudentTestFile();
        writeToCSV();
    }

    private void writeToStudentDetailsFile() {
        try (BufferedWriter writer = new BufferedWriter(new FileWriter("A:\\oop\\Project\\objs\\StudentDetails.txt", true))) {
            writer.write(this.rollNum + "," + this.cgpa + "," + this.testScore + "," + this.skills + "," + this.expectedCTC +"\n");
        } catch (IOException e) {
            System.out.println("An error occurred while writing to StudentDetails.txt.");
            e.printStackTrace();
        }
    }

    private void writeToStudentTestFile() {
        try (BufferedWriter writer = new BufferedWriter(new FileWriter("A:\\oop\\Project\\objs\\StudentTest.txt", true))) {
            writer.write(this.rollNum + ":" + this.testScore + "\n");
        } catch (IOException e) {
            System.out.println("An error occurred while writing to StudentTest.txt.");
            e.printStackTrace();
        }
    }

    private void writeToCSV() {
        try (BufferedWriter writer = new BufferedWriter(new FileWriter("A:\\oop\\Project\\objs\\StudentAca.csv", true))) {
            writer.write(this.rollNum + "," + this.cgpa + "," + this.testScore + "," + this.skills + "," + this.expectedCTC +  "\n");
        } catch (IOException e) {
            System.out.println("An error occurred while writing to StudentAca.csv.");
            e.printStackTrace();
        }
    }

    public String getRollNum() {
        return rollNum;
    }

    public double getTestScore() {
        return testScore;
    }

    public float getCgpa() {
        return cgpa;
    }

    public String getSkills() {
        return skills;
    }
    public float getExpectedCTC() {
        return expectedCTC;
    }
}
